### PR TITLE
[ONNX] Use TracedONNXFunction op signature to promote inputs to tensors

### PR DESCRIPTION
**Pull request by titaiwangms**
Previous to this PR, in torchlib TracedONNXFunction, the inputs could be python constants even if the annotation sets to TensorTypes.

### CODE REVIEW

File: torch/onnx/_internal/exporter/_building.py [hunk 1]
```diff
@@ -660,6 +660,32 @@ def eval_function(  # type: ignore[override]
...
+                        self.opset,
+                    )
+
+                except Exception as e:
+                    raise _errors.GraphConstructionError(
+                        f"Error processing Python constants for operator '{op_signature.domain}::{op_signature.name}'. "
+                        f"named_inputs={named_inputs}, named_attrs={named_attrs}, opset={self.opset}, op_signature={op_signature}."
+                    ) from e
+
                 return function.function(**named_inputs, **named_attrs)
```
Comment (justinchuby) [inline torch/onnx/_internal/exporter/_building.py:689]: update?

**Comment by titaiwangms** [comment]
@justinchuby I totally forgot the status of this. Are we still doing this?

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
One sec

**Comment by justinchuby** [comment]
Maybe just do the constants for now? What do you think?

**Comment by titaiwangms** [comment]
I will test more on this.

**Comment by titaiwangms** [comment]
> Maybe just do the constants for now? What do you think?

Agree. python list related properties are leveraged heavily in trace_only function. Also, `_process_python_constant` is sufficient to solve the symint (fx.Node) and int issue.

**Pull request closed without merging**